airbyte-integrations/bases/base-java/src/test/java/io/airbyte/integrations/base/IntegrationRunnerTest.java

```diff
@@ -377,4 +378,48 @@ void testParseConnectorImage() {
     assertEquals("1.0.1-alpha", IntegrationRunner.parseConnectorVersion(":1.0.1-alpha"));
   }
 
+  @Test
+  void testConsumptionOfInvalidStateMessage() {
+    final String invalidStateMessage = """
+                                       {
+                                         "type" : "STATE",
+                                         "state" : {
+                                           "type": "NOT_RECOGNIZED",
+                                           "global": {
+                                             "streamStates": {
+                                               "foo" : "bar"
+                                             }
+                                           }
+                                         }
+                                       }
+                                       """;
+
+    Assertions.assertThrows(IllegalStateException.class, () -> {
+      try (final AirbyteMessageConsumer consumer = mock(AirbyteMessageConsumer.class)) {
+        IntegrationRunner.consumeMessage(consumer, invalidStateMessage);
+      }
+    });
+  }
+
+  @Test
+  void testConsumptionOfInvalidNonStateMessage() {
+    final String invalidNonStateMessage = """
+                                          {
+                                            "type" : "NOT_RECOGNIZED",
```

Comment:
Why does this test doesn't throw, it shouldn't match any enum value and not be able to properly deserialize it.

Reply:
@benmoriceau The first call to `Jsons.tryDeserialize` swallows the exception internally and returns an empty `Optional`.  This is the root of the problem, as it means that the existing code just continues on when it fails to deserialize the JSON into the `AirbyteMessage`.  This particular test proves the existing behavior for non-state messages is preserved, per the discussion with @davinchia on the linked issue  #14067),

Reply:
See https://github.com/airbytehq/airbyte/issues/14067#issuecomment-1165549549.

Generally we don't want to kill the entire sync if we have one malformed record. We do want to kill the sync if the state message is malformed because that affects sync correctness.

Reply:
hah we replied at the same time!